### PR TITLE
Add types for junit-report-builder

### PR DESCRIPTION
**Pull request by dakoenig**
This PR adds types for the previously untyped junit-report-builder package.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/junit-report-builder/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+// Type definitions for junit-report-builder 3.0
+// Project: https://github.com/davidparsson/junit-report-builder
+// Definitions by: Daniel König <https://github.com/dakoenig>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace builder {
+    interface Builder {
```
Comment (peterblazejewicz) [inline types/junit-report-builder/index.d.ts:7]: the symbol is named JUnitReportBuilder apparently:
```ts
JUnitReportBuilder { _factory: Factory {}, _testSuitesAndCases: [] }
```

File: types/junit-report-builder/junit-report-builder-tests.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    .standardOutput('output written to stdout')
+    .standardError('output written to stderr')
+    .errorAttachment('/path/to/attachment');
+
+// $ExpectType string
+builder.build();
+// $ExpectType void
+builder.writeTo('test-report.xml');
+
+// $ExpectType Builder
```
Comment (peterblazejewicz) [inline types/junit-report-builder/junit-report-builder-tests.ts:53]: _ditto_, the symbol name is different: `JUnitReportBuilder { _factory: Factory {}, _testSuitesAndCases: [] }`

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@dakoenig thanks!

**Comment by dakoenig** [comment]
Ready to merge

**Pull request merged**